packages/map/src/leaflet/Icons.ts

```diff
@@ -0,0 +1,127 @@
+                        marker.bindTooltip(tooltipText, {
+                            direction: "top",
+                            offset: point(0, -34)
+                        });
+                    }
+                }
+                this.add(marker);
+            });
+        }
+    }
+
+    _currSelRow;
+    clickHandler(e, marker, row) {
+        const sel = this._selection.click(marker._icon);
+        this._currSelRow = sel ? row : undefined;
+        this.click(this.rowToObj(row), "", sel);
+    }
+
+    //  Events  ---
+    click(row, col, sel) {
+    }
+}
+Icons.prototype._class += " map_Icons";
+
+export interface Icons {
+    latitudeColumn(): string;
+    latitudeColumn(_: string);
+    longtitudeColumn(): string;
+    longtitudeColumn(_: string);
+    omitNullLatLong(): boolean;
+    omitNullLatLong(_: boolean);
+    iconUrl(): string;
+    iconUrl(_: string);
+    iconUrl_exists(): boolean;
+    iconUrlColumn(): string;
+    iconUrlColumn(_: string);
+    iconUrlColumn_exists(): boolean;
+    iconHtml(): string;
+    iconHtml(_: string);
+    iconHtml_exists(): boolean;
+    iconHtmlColumn(): string;
+    iconHtmlColumn(_: string);
+    iconHtmlColumn_exists(): boolean;
+    iconWidth(): number;
+    iconWidth(_: number);
+    iconHeight(): number;
+    iconHeight(_: number);
+    iconAnchorX(): number;
+    iconAnchorX(_: number);
+    iconAnchorY(): number;
+    iconAnchorY(_: number);
+    tooltipColumn(): string;
+    tooltipColumn(_: string);
+    tooltipColumn_exists(): boolean;
+}
+
+Icons.prototype.publish("latitudeColumn", null, "set", "Latitude column", function () { return this.columns(); }, { optional: true });
+Icons.prototype.publish("longtitudeColumn", null, "set", "Longtitude column", function () { return this.columns(); }, { optional: true });
+Icons.prototype.publish("omitNullLatLong", true, "boolean", "Remove lat=0,lng=0 from IconsData", null, { tags: ["Basic"] });
+Icons.prototype.publish("iconUrl", "", "string", "Icon URL", null); // , { disable: (w: Icons) => w.iconHtml_exists() || w.iconHtmlColumn_exists() });
+Icons.prototype.publish("iconUrlColumn", null, "set", "Icon URL column", function () { return this.columns(); }, { optional: true });
+Icons.prototype.publish("iconHtml", "", "string", "Icon html", null); // , { disable: (w: Icons) => w.iconUrl_exists() || w.iconUrlColumn_exists() });
+Icons.prototype.publish("iconHtmlColumn", null, "set", "Icon html column", function () { return this.columns(); }, { optional: true });
+Icons.prototype.publish("iconWidth", 32, "number", "Icon width");
+Icons.prototype.publish("iconHeight", 32, "number", "Icon height");
+Icons.prototype.publish("iconAnchorX", 16, "number", "Icon Anchor X");
+Icons.prototype.publish("iconAnchorY", 32, "number", "Icon Anchor Y");
+Icons.prototype.publish("tooltipColumn", null, "set", "Tooltip column", function () { return this.columns(); }, { optional: true });
+
+export class ClusterIcons extends Icons {
```

Comment:
Why not create a new `ClusterIcons.ts`  file for this?

Reply:
Because its 5 lines long.